apollo-router/src/query_planner/dual_query_planner.rs

```diff
@@ -242,6 +240,31 @@ pub fn plan_matches(js_plan: &QueryPlanResult, rust_plan: &QueryPlan) -> bool {
     opt_plan_node_matches(js_root_node, &rust_root_node)
 }
 
+pub fn diff_plan(js_plan: &QueryPlanResult, rust_plan: &QueryPlan) -> String {
+    let js_root_node = &js_plan.query_plan.node;
+    let rust_root_node = convert_root_query_plan_node(rust_plan);
```

Comment:
Both of these parameters _can_ be `Option<PlanNode>`, but that would require additionally exporting `convert_root_query_plan_node` as part of `_private`, and I want to limit that amount of things we export, even though they are under `_private`.